Feature/152798641/redesign authentication page

zeze111 opened the pull request:
<img width="1440" alt="screen shot 2017-12-07 at 12 04 24 pm" src="https://user-images.githubusercontent.com/31539842/33712490-f52d09ea-db47-11e7-8623-894d29f1e6d4.png">
#### What does this PR do?
Redesigns the UI of Sign up and Sign in page

#### Description of Task to be completed?
 * write the html and css code for the sign up and sign in page
 * remove existing code in the signup.html, signing.html and replace with new html code written
 * remove existing style sheet and replace with the custom css code written
 * include the stylesheet in the `head.jade` file

#### How should this be manually tested?
Have the app working by:
 * run the command `mongod` on the command line to start the mongo server
 * run command `yarn start` on the command line to start the app server
 * once app is running, navigate to the app using `localhost:3000`
 * navigate to Sign up or Sign in page

 #### Any background context you want to provide?
 Nil

 #### What are the relevant pivotal tracker stories?
 #152798641

 #### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2017-12-07 at 12 04 24 pm" src="https://user-images.githubusercontent.com/31539842/33717176-3a09ccf4-db59-11e7-8a2a-5af58734e010.png">

![Uploading Screen Shot 2017-12-07 at 12.04.47 PM.png…]()

Commit:
resolve conflicts

Commit:
resolve conflict

Commit:
feat(Views): redesign UI for authentication pages
- re- design sign up page
- re-design sign in page
[Delivers #152798641]

Commit:
feat(Views): redesign UI for authentication pages
- re- design sign up page
- re-design sign in page
[Delivers #152798641]

zeze111 committed:
Merge branch 'staging' into feature/152798641/redesign-authentication-page

sachungo commented on package.json:
```diff
@@ -1,95 +1,97 @@
-    "should": "^13.1.3",
-    "supertest": "^3.0.0"
-  }
+    "name": "cardsforhumanity",
```
The package.json should be using 2 spaces for a tab instead of 4 spaces. The spacing of the deletions in the file was ok.

Always, choose `incoming changes or both` when fixing conflicts and ensure everything works well together. The deletions have `babel`, `babel-cli`, `gulp-babel` etc that was added to the package.json from staging but missing in this branch.

Revert to staging package.json and then add the new package you added  in this branch

sachungo commented on public/views/signin.html:
```diff
@@ -1,71 +1,91 @@
+  <nav class="navbar navbar-expand-sm navbar-light navbar-inverse bg-light fixed-top ">
+    <a class="navbar-brand" href="/" style="cursor:pointer">
+      <span id="cards">Cards</span> For Humanity</a>
+    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
```
If a tag has many attribute, it can be written as
```html
<button
  class="navbar-toggler"
  type="button"
  data-toggle="collapse"
  data-target="#navbarSupportedContent"
  aria-controls="navbarSupportedContent"
  aria-expanded="false"
  aria-label="Toggle navigation"
>
```

The tab space in the above html corresponds to 2 spaces

sachungo commented on public/views/signin.html:
```diff
@@ -1,71 +1,91 @@
+  <div>
+    <div id="" class="container-fluid" style="margin:0 auto; height: 90vh">
+      <div class="row" style="padding-top: 8em">
+        <!-- <div class="col-md-6">
```
If this div isn't being used, then delete it instead of commenting it out

sachungo commented on public/views/signin.html:
```diff
@@ -1,71 +1,91 @@
+                <div style="padding-right: 4em;">
+                  <div class="form-group">
+                    <label class=" text-right col-md-3" for="email">User:</label>
+                    <input class="col-md-5 input-text" type="email" class="form-control" 
```
Same as the comment with the `<button>` example. 
```html
<input
  class="col-md-5
  input-text"
  type="email"
  class="form-control"
  .....
  .....
>

sachungo commented on public/views/signin.html:
```diff
@@ -1,71 +1,91 @@
   </div>
-</div>
-
+</div>
```
Add new line to end of file. It will be much easier to set your editor to add it automatically when saving. If using Visual Studio Code, then add this to your settings:
```json
{
   "files.trimTrailingWhitespace": true,
   "files.insertFinalNewline": true,
   "files.trimFinalNewlines": true 
 }

sachungo commented on public/views/signup.html:
```diff
@@ -1,86 +1,94 @@
+  <nav class="navbar navbar-expand-sm navbar-light navbar-inverse bg-light fixed-top ">
+    <a class="navbar-brand" href="/" style="cursor:pointer">
+      <span id="cards">Cards</span> For Humanity</a>
+    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
```
Same as the button comment

sachungo reviewed (changes requested):
I have only reviewed the code and not tested. I'm unable to test locally as the branch doesn't have the packages added for babel so `yarn start` fails. I have left a comment on what to do in `package.json` feedback

Commit:
resolve conflicts

Commit:
feature(views): Implemented feedback
- refactored code according to standrads
- resolved missing packages in package.json

sachungo commented on public/views/signin.html:
```diff
@@ -1,71 +1,158 @@
+    <a
+      class="navbar-brand"
+      href="/"
+      style="cursor:pointer">
```
`>` for multiline attributes should always be on a new line. I.e.
```html
<a
  class="navbar-brand"
  href="/"
  style="cursor:pointer"
>
  <span id="cards">Cards</span>
  For Humanity
</a>

sachungo reviewed (approved):
Tested locally and it works.
Are the input fields for signup/sign-in page different in the design? 
Also, should the avatars be aligned with the design? The last 2 are not aligned with the others.

<img width="563" alt="screen shot 2017-12-14 at 8 40 42 am" src="https://user-images.githubusercontent.com/12892171/33977233-f6c62436-e0aa-11e7-82de-9a537cc6ef3f.png">

<img width="519" alt="screen shot 2017-12-14 at 8 40 50 am" src="https://user-images.githubusercontent.com/12892171/33977241-ff54104a-e0aa-11e7-88d1-3367a978a9db.png">

I have left 1 comment which should be taken into account for future implementations.
@zeze111 left a comment and questions to be clarified before it's merged

Commit:
feature(views)- fix sign up/in page
- corrected design for sign in input fields
- corrected design for avatars on sign up page

Commit:
feature(views)- fix sign up/in page
- corrected design for sign in input fields
- corrected design for avatars on sign up page
- added validation errors

Commit:
fix merge conflict

sachungo commented on public/js/controllers/index.js:
```diff
@@ -29,24 +29,33 @@ angular.module('mean.system')
+        $scope.email &&
+        $scope.password
+      ) {
+        var userData = {
```
use `let`

sachungo reviewed (commented):
@zeze111 the second input box for sign in should have a `Password` placeholder.
![image](https://user-images.githubusercontent.com/12892171/34027861-7f4a0352-e170-11e7-971c-ac9e3d39827c.png)

Also, fix the failing Travis test(s) and the PR will be good to go

Great work 👍

Commit:
feat(views) : fix typos
- fix typo errors in sign in page
- use of correct variables declarartion
